nomenclature/processor/data_validator.py

```diff
@@ -276,6 +287,37 @@ def from_file(
 
         return cls(file=file, criteria_items=criteria_items, output_path=output_path)  # type: ignore
 
+    @classmethod
+    def from_codelist(
+        cls, codelist: VariableCodeList, output_path: Path | str | None = None
+    ) -> "DataValidator":
+        criteria_items = []
+
+        for variable in codelist.values():
+            if any(
+                [
+                    attribute in VALIDATION_ARGS
+                    for attribute in variable.extra_attributes
+                ]
+            ):
+                criteria_items.append(
+                    dict(
+                        variable=variable.name,
+                        validation=[
+                            dict(
+                                [
+                                    (key, value)
+                                    for key, value in variable.extra_attributes.items()
+                                    if key in VALIDATION_ARGS
+                                ]
+                            )
+                        ],
+                    )
+                )
+        return cls(
+            file="definitions", criteria_items=criteria_items, output_path=output_path
+        )  # type: ignore
```

Comment:
I think the reason the type checker complains is because the type hint in the function inputs is incorrect. I would fix that rather than tell the type checker to ignore it.